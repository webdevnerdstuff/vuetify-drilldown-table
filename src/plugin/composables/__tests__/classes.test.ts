import { describe, it, expect } from 'vitest';
import {
	useTableClasses,
	useCellAlignClasses,
	useCheckBoxClasses,
	useCellClasses,
	useHeaderRowClasses,
	useHeaderCellClasses,
	useSortIconClasses,
	useBodyCellClasses,
	useBodyRowClasses,
	useTFootClasses,
	useTFootRowClasses,
	useTFootCellClasses,
} from '../classes';


describe('Classes Composable', () => {

	describe('useTableClasses', () => {
		it('should return class object', () => {
			const classes = useTableClasses({
				elevation: 5,
				fixedHeader: true,
				isDrilldown: true,
				isHover: true,
				isServerSide: true,
				level: 1,
				mobile: false,
				separator: true,
			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useCellAlignClasses', () => {
		it('should return class object', () => {
			const classes = useCellAlignClasses({});
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useCellAlignClasses({ align: 'start' });
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useCheckBoxClasses', () => {
		it('should return class object', () => {
			const classes = useCheckBoxClasses({ level: 0 });
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useCheckBoxClasses({ level: 1 });
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useCellClasses', () => {
		it('should return class object', () => {
			const classes = useCellClasses({
				column: {},
				elm: 'header',
				level: 0,
			});
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useCellClasses({
				column: { cellClass: 'foo-cell' },
				elm: 'header',
				level: 0,
			});
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useHeaderRowClasses', () => {
		it('should return class object', () => {
			const classes = useHeaderRowClasses({ level: 0 });
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useHeaderRowClasses({ level: 1 });
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useHeaderCellClasses', () => {
		it('should return class object', () => {
			const classes = useHeaderCellClasses({
				column: {
					align: 'start',
					key: 'name',
					title: 'Name',
				},
				level: 0,
				slotName: 'header-data-table-select',
			});
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useSortIconClasses', () => {
		it('should return class object', () => {
			const classes = useSortIconClasses({
				iconOptions: {
					defaultSet: 'mdi',
					id: 'id',
					level: 0,
					sortBy: [{
						key: 'id',
						order: 'asc',
					}],
				},
				key: 'id',
				level: 0,
				sortBy: [{
					key: 'id',
					order: 'desc',
				}],
			});
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useBodyCellClasses', () => {
		it('should return class object', () => {
			const classes = useBodyCellClasses({
				column: {
					align: 'start',
					key: 'name',
					title: 'Name',
				},
				level: 0,
			});
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useBodyRowClasses', () => {
		it('should return class object', () => {
			const classes = useBodyRowClasses({
				expandOnClick: true,
				level: 0,
				levels: 2,
			});
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useBodyRowClasses({
				expandOnClick: false,
				level: 0,
				levels: 2,
			});
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useBodyRowClasses({
				expandOnClick: true,
				level: 2,
				levels: 1,
			});
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useTFootClasses', () => {
		it('should return class object', () => {
			const classes = useTFootClasses({ level: 0 });
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useTFootClasses({ level: 1 });
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useTFootRowClasses', () => {
		it('should return class object', () => {
			const classes = useTFootRowClasses({ level: 0 });
			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useTFootRowClasses({ level: 1 });
			expect(classes).toMatchSnapshot();
		});
	});

	describe('useTFootCellClasses', () => {
		it('should return class object', () => {
			const classes = useTFootCellClasses({
				column: {
					align: 'start',
					key: 'name',
					title: 'Name',
				},
				level: 0,
				slotName: 'tfoot-data-table-expand'
			});
			expect(classes).toMatchSnapshot();
		});
	});
});
