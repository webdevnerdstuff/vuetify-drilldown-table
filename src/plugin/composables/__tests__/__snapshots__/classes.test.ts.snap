// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Classes Composable > useBodyCellClasses > should return class object 1`] = `
{
  "undefined": false,
  "v-drilldown-table--body-row-td": true,
  "v-drilldown-table--body-row-td-0": true,
}
`;

exports[`Classes Composable > useBodyRowClasses > should return class object 1`] = `
{
  "v-data-table__tr": true,
  "v-data-table__tr--clickable": true,
  "v-data-table__tr--mobile": undefined,
  "v-drilldown-table--body-row": true,
  "v-drilldown-table--body-row-0": true,
}
`;

exports[`Classes Composable > useBodyRowClasses > should return class object 2`] = `
{
  "v-data-table__tr": true,
  "v-data-table__tr--clickable": false,
  "v-data-table__tr--mobile": undefined,
  "v-drilldown-table--body-row": true,
  "v-drilldown-table--body-row-0": true,
}
`;

exports[`Classes Composable > useBodyRowClasses > should return class object 3`] = `
{
  "v-data-table__tr": true,
  "v-data-table__tr--clickable": false,
  "v-data-table__tr--mobile": undefined,
  "v-drilldown-table--body-row": true,
  "v-drilldown-table--body-row-2": true,
}
`;

exports[`Classes Composable > useCellAlignClasses > should return class object 1`] = `
{
  "d-flex align-center": true,
  "justify-start": true,
  "justify-undefined": false,
}
`;

exports[`Classes Composable > useCellAlignClasses > should return class object 2`] = `
{
  "d-flex align-center": true,
  "justify-start": false,
}
`;

exports[`Classes Composable > useCellClasses > should return class object 1`] = `
{
  "undefined": false,
  "v-drilldown-table--header-row-td": true,
  "v-drilldown-table--header-row-td-0": true,
  "v-drilldown-table--header-row-td-mobile": undefined,
}
`;

exports[`Classes Composable > useCellClasses > should return class object 2`] = `
{
  "foo-cell": "foo-cell",
  "v-drilldown-table--header-row-td": true,
  "v-drilldown-table--header-row-td-0": true,
  "v-drilldown-table--header-row-td-mobile": undefined,
}
`;

exports[`Classes Composable > useCheckBoxClasses > should return class object 1`] = `
{
  "d-flex": true,
  "v-drilldown-table--header-select-all-checkbox": true,
  "v-drilldown-table--header-select-all-checkbox-0": true,
}
`;

exports[`Classes Composable > useCheckBoxClasses > should return class object 2`] = `
{
  "d-flex": true,
  "v-drilldown-table--header-select-all-checkbox": true,
  "v-drilldown-table--header-select-all-checkbox-1": true,
}
`;

exports[`Classes Composable > useHeaderCellClasses > should return class object 1`] = `
{
  "undefined": false,
  "v-drilldown-table--header-row-th": true,
  "v-drilldown-table--header-row-th-0": true,
  "v-drilldown-table--header-row-th-header-data-table-select": true,
  "v-drilldown-table--header-row-th-header-data-table-select-0": true,
  "v-drilldown-table--header-row-th-sortable": false,
  "v-drilldown-table--header-row-th-sortable-default-color": false,
}
`;

exports[`Classes Composable > useHeaderRowClasses > should return class object 1`] = `
{
  "v-drilldown-table--header-row": true,
  "v-drilldown-table--header-row-0": true,
}
`;

exports[`Classes Composable > useHeaderRowClasses > should return class object 2`] = `
{
  "v-drilldown-table--header-row": true,
  "v-drilldown-table--header-row-1": true,
}
`;

exports[`Classes Composable > useSortIconClasses > should return class object 1`] = `
{
  "fa-fw": false,
  "mx-1": true,
  "v-drilldown-table--header-row-th-sortable-sort-icon": true,
  "v-drilldown-table--header-row-th-sortable-sort-icon-0": true,
  "v-drilldown-table--header-row-th-sortable-sort-icon-asc": false,
  "v-drilldown-table--header-row-th-sortable-sort-icon-desc": true,
}
`;

exports[`Classes Composable > useTFootCellClasses > should return class object 1`] = `
{
  "undefined": false,
  "v-drilldown-table--tfoot-row-td": true,
  "v-drilldown-table--tfoot-row-td-0": true,
  "v-drilldown-table--tfoot-row-td-tfoot-data-table-expand": true,
  "v-drilldown-table--tfoot-row-td-tfoot-data-table-expand-0": "tfoot-data-table-expand",
}
`;

exports[`Classes Composable > useTFootClasses > should return class object 1`] = `
{
  "v-drilldown-table--tfoot": true,
  "v-drilldown-table--tfoot-0": true,
}
`;

exports[`Classes Composable > useTFootClasses > should return class object 2`] = `
{
  "v-drilldown-table--tfoot": true,
  "v-drilldown-table--tfoot-1": true,
}
`;

exports[`Classes Composable > useTFootRowClasses > should return class object 1`] = `
{
  "v-drilldown-table--tfoot-row": true,
  "v-drilldown-table--tfoot-row-0": true,
}
`;

exports[`Classes Composable > useTFootRowClasses > should return class object 2`] = `
{
  "v-drilldown-table--tfoot-row": true,
  "v-drilldown-table--tfoot-row-1": true,
}
`;

exports[`Classes Composable > useTableClasses > should return class object 1`] = `
{
  "elevation-5": true,
  "v-drilldown-table": true,
  "v-drilldown-table--child": true,
  "v-drilldown-table--fixed-header": true,
  "v-drilldown-table--hover": true,
  "v-drilldown-table--level-1": true,
  "v-drilldown-table--mobile": false,
  "v-drilldown-table--separator-true": true,
  "v-drilldown-table--server": true,
}
`;
